app.py

```diff
@@ -23,4 +38,25 @@ def q1(g, w):
     g = Graph()
     w = generate_weights(g.E)
     g.print_graph(w)
-    q1(g, w)
+    r = g.V[random.randint(0, len(g.V) - 1)]  # pick random start
+    q1(g, w, r)
+
+    src = random.randint(0, N - 1)
+    dest = random.randint(0, N - 1)
+    while (dest in g.adj[src] or src == dest):           
+        src = random.randint(0, N - 1)
+        dest = random.randint(0, N - 1)     
+    
+    # make new edge the most 'expensive' for both veritces
+    new_w = max(max_with_inf(x) for x in w)  + 1 
+    new_edge(g, w, src, dest, new_w)
+    print("First try: \n The new edge is " + str(Edge(src,dest)) + " and thie weight is " + str(new_w))
+    q2(g, src, dest, w)
+    print_mst(g, r)
+
+    # make new edge the most 'cheap' for both veritces
+    new_w = min(w[src] + w[dest]) - 1
```

Comment:
Doesnt seem to work. Why not the cheapest weight in graph (say 0)?
You can do this by min([min([x for x in a if x!=float('inf')]) for a in w])

Reply:
we can do that, but its work

min([min([x for x in a if x!=float('inf')]) for a in w]) not needed
only the src V and the dest V weights be less than the new one

Reply:
change to 0 ?

and change the max to 100?

Reply:
done